Proper battery sense fix

Vyolex opened the pull request:
This commit properly "fixes" the battery sense, in the previous fix I submitted (#139) I simply swapped the R1 and R2 values not noticing that the formula itself was wrong, not the default values. 

This rectifies that mistake by swapping the values again and fixing the formula instead. Now it should be consistent with the comments and the actual values inside the datasheet again and should cause less confusion.

This fix is OPTIONAL as it does not change anything relating to the behavior and only tries to improve code readability and consistency.  

![image](https://user-images.githubusercontent.com/25586367/214426139-79c1b257-50d6-46c1-a31e-9435786b56be.png)

Eirenliel commented:
also i'm thinking here... production slimes have different ADC resistors and the firmware is the same for them... um...

![production-adc](https://user-images.githubusercontent.com/1257939/214429823-2125ff87-520d-4321-9221-b01b23b7f197.png)

Vyolex commented:
Technically it doesn't matter if the ratio are the same. The common configuration on DIY would equate to 400 and 100 so the 40.2 and 10 should be close enough. But that might be something we should fix, even if only for readability. I'm fairly certain I mentioned in my last Pr that these really should be in the defines or definitions file for that exact reason :p

unlogisch04 commented on src/batterymonitor.h:
```diff
@@ -62,7 +62,7 @@
     // Diagramm: 
     //   (Battery)--- [BATTERY_SHIELD_RESISTANCE] ---(INPUT_BOARD)---  [BATTERY_SHIELD_R2] ---(ESP_INPUT)--- [BATTERY_SHIELD_R1] --- (GND)
```
Your diagram is different than this one here.
I don't think we should change anything.
There is not clear definition for R1 and R2 how it have to be.

Vyolex replied:
> Wemos D1 Mini has an internal Voltage Divider with R1=220K and R2=100K > this means, 3.3V analogRead input voltage results in 1023.0

Then this line has conflicting information, I agree that it is just naming and doesn't really matter. I just wanted to make it more consistent with documentation that is available. I am completely fine reverting all changes and just adapting the values in this single line above though.

Eirenliel commented:
> Technically it doesn't matter if the ratio are the same. The default defines as they stand would equate to 400 and 100 so the 40.2 and 10 should be close enough. But that might be something we should fix, even if only for readability. I'm fairly certain I mentioned in my last Pr that these really should be in the defines or definitions file for that exact reason :p

but they aren't the same we don't change firmware for battery sense before uploading 😭

Vyolex commented:
> but they aren't the same we don't change firmware for battery sense before uploading 😭

I know we don't change the values, but the ratio is what is important;

(40.2+10)/10 = 5.02
(400 + 100)/100 = 5

The difference is extremely minimal (only 0.4%), idk what deviation is allowed on the resistors used within the official trackers but this is much less than the recommended 1% for DIY use. 

That being said I fully agree if you would change it for the future slimes, I'm just trying to stress that it is not a very big issue.

Eirenliel commented:
oh right i forgor that it adds +180, so we gucci.

Vyolex commented:
Reverted earlier changes but adapted comments so everything is consistent.

Added board specific resistor values with override option (and fallback).

Eirenliel commented:
What I'm not sure about is that NODEMCU actually doesn't have battery shield.
